### ISSUE TITLE
Use current user location in agents

### ISSUE THREAD

**Issue by Azimoth** [huginn/huginn#2073]
I tried to set up a weather agent for my current location rather than for one specific location.

So far, I managed to get my location via mqtt in a JSON string (?).

{
"topic": "owntracks/",
"message": {
"_type": "location",
"tid": "3g",
"t": "u",
"conn": "w",
"lat": 66.6,
"lon": 66.6,
"tst": 1234567890,
"acc": 10,
"batt": 99
},
"time": 1234567890
}

How would I get the lat and lon from this event into the location field of the weather agent?
Or: How would I make the location from this available to all agents? Is there a general "user location attribute" that could be requested by all agents? (If not: How hard would it be to create this?)

**Comment by dsander** [comment]
Are you using darksky or wunderground in the `WeatherAgent`? If it is darksky you can use a `CommanderAgent` with the `configure` action and set the `location` to your coordinates with liquid `{{lon}},{{lat}}`. For wunderground you first need to convert the coordinates to a to a location their API supports.

**Comment by Azimoth** [comment]
I noticed that, when I read the agent descriptions, so I chose darksky right from the start.

Now that you pointed me to the CommanderAgent, I saw that a case quite similar to mine is used as an example. Nevertheless I'm a littls stuck.

Sources: MQTT Agent

{
action:configure
configure_options:{ 'location': '{{_location_.latlng}}' }
}

Ok. But this lead to an error. Where should I put this sample MQTT string from the lower right corner in the CommandAgent description?

{
  "topic": "owntracks/kcqlmkgx/Dan",
  "message": {"_type": "location", "lat": "-34.8493644", "lon": "138.5218119", "tst": "1401771049", "acc": "50.0", "batt": "31", "desc": "Home", "event": "enter"},
  "time": 1401771051
}

**Comment by dsander** [comment]
If your MQTT events look like this those options should work:
```json
{
  "action": "configure",
  "configure_options": {
    "location": "{{message.lon}},{{message.lat}}"
  }
}
```

**Comment by Azimoth** [comment]
Well... using quotes like you did made the error go away. -.-' It works now, thank you!

If I hadn't asked for help here, how could I have found out about this? Did I not find available documentation? Or would I have to learn Ruby first (or Liquid, which I had never heard of before)?

**Comment by dsander** [comment]
You are right the documentation is a bit sparse. You don't really need to know ruby, reading up on JSON (the format in which the Agent options are stored) and Liquid (we have some information on Huginn specifics the [wiki](https://github.com/huginn/huginn/wiki/Formatting-Events-using-Liquid)) should be a good start.
